### PR TITLE
dev/fix no use before define lint violations

### PR DESCRIPTION
**Pull request by rjchow**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Fixed all instances of `@typescript-eslint/no-use-before-define` violations

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

While this PR introduces some code changes, there should be no logical change as the changes all involve moving of function declarations only. Thus, there should be no observable changes to the application.

For safety, these are the features that saw changes:
- Activity Panel 
- Analytics, Historical data query
- Homescreen tasklist
- Homescreen, WC Admin layout
- Homescreen store management links (could be dead code? I don't see it on the homescreen)
- Marketing, install and load plugins
- Marketplace, feedback modal

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Fixed all violations of no-use-before-define eslint rule

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/activity-panel/activity-panel.js [hunk 1]
```diff
@@ -73,6 +73,19 @@ export const ActivityPanel = ( { isEmbedded, query } ) => {
 	const { updateUserPreferences, ...userData } = useUserPreferences();
 	const activeSetupList = useActiveSetupTasklist();
 
+	const closePanel = () => {
```
Comment (crunnells) [inline plugins/woocommerce-admin/client/activity-panel/activity-panel.js:76]: Was this causing an error, or just a "declare the function before you use it" kind of change?

**Review by crunnells — APPROVED** [review]
I get it now; no-use-before-define is why so much code was moved around. Surprised it even worked in the first place! Also, more great QOL changes for translators. LGTM!

**Review by psealock — APPROVED** [review]
Looks good to me 🚢

**Comment by rjchow** [comment]
I think we could consider a smoke test of the affected features:

    Activity Panel
    Analytics, Historical data query
    Homescreen tasklist
    Homescreen, WC Admin layout
    Homescreen store management links (could be dead code? I don't see it on the homescreen)
    Marketing, install and load plugins
    Marketplace, feedback modal

**Pull request merged**